scripts/dev/upstream_merge/upstream_merge_and_test.py

```diff
@@ -0,0 +1,333 @@
+import datetime
+import os
+import argparse
+import json
+import logging
+from GitRepo import *
+from git_commands import send_email
+from build import build_images
+from test import OS_test
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Automated repository merging script")
+    parser.add_argument("-c", type=str, help="Path to configuration file", default="scripts/dev/upstream_merge/automation_conf.json")
+    parser.add_argument("-skip-merge", type=bool, help="Skip merging with upstream", default=False)
+    args = parser.parse_args()
+
+    return args
+
+def parse_config_file(config_file_path):
+    try:
+        with open(config_file_path, "r") as file:
+            config = json.load(file)
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        print(f"Error loading config file: {e}")
+        exit(1)
+    
+    return config.get("NILRT_branch"), os.getcwd()+f"/{config.get('conf_file_path')}", config.get("force_checkout"), config.get("username"), config.get("upstream_repo_name"), config.get("merge_branch_name"), config.get("email_from"), config.get("email_to"), config.get("email_log_level"), config.get("log_level"), config.get("work_item_id"), config.get("VM_name"), config.get("snapshot_name")
```

Comment:
Would it make sense to create a config object instead of passing a huge tuple back?